client/my-sites/people/people-section-header/invite-form.jsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import Debug from 'debug';
+import LinkedStateMixin from 'react-addons-linked-state-mixin';
+
+/**
+ * Internal Dependencies
+ */
+import RoleSelect from 'my-sites/people/role-select';
+import TokenField from 'components/token-field';
+import Button from 'components/button';
+import { sendInvites } from 'lib/invites/actions';
+
+/**
+ * Module Variable
+ */
+const debug = new Debug( 'calypso:invite-people-form' );
+
+export default React.createClass( {
+	displayName: 'InviteForm',
+	mixins: [ LinkedStateMixin ],
+
+	componentWillReceiveProps() {
+		this.setState( this.resetState() );
+	},
+
+	getInitialState() {
+		return this.resetState();
+	},
+
+	resetState() {
+		return ( {
+			usernamesOrEmails: [],
+			role: '',
+			message: '',
+			sendingInvites: false
+		} );
+	},
+
+	onTokensChange( tokens ) {
+		this.setState( { usernamesOrEmails: tokens } );
+	},
+
+	submitForm() {
+		debug( 'Sending invites.', this.state );
+		this.setState( { sendingInvites: true } );
+		sendInvites( this.props.site.ID, this.state.usernamesOrEmails, this.state.role, this.state.message, this.handleResponse );
+	},
+
+	handleResponse( error, data ) {
+		debug( 'Request to send invites complete. Results:', error, data );
+		this.setState( this.resetState() );
+	},
+
+	render() {
+		return (
+			<div className="invite-form">
```

Comment:
I think the display name should be `PeopleSectionHeaderInviteForm` and the class should be `people-section-header__invite-form`. This is a bit awkward since it's a separate component, but it's meant to be used as a child of `people-section-header`.